app/src/components/Map/ControlPanel.jsx

```diff
@@ -32,30 +32,52 @@ class ControlPanel extends Component {
     }
   }
 
+  calculateVesselCount() {
+    const min = this.props.timelineInnerExtent[0].getTime();
+    const max = this.props.timelineInnerExtent[1].getTime();
+
+    if (this.props.chartData.length === 0) {
+      return '-';
+    }
+
+    let result = this.props.chartData.reduce((acc, elem) => {
+      if (elem.date >= min && elem.date <= max) {
+        return acc + elem.value;
```

Comment:
To me, this is computed view data, not real app state. Adding this to the state container is an implicit state duplication, which is what we're trying to avoid.